src/controller.ts

```diff
@@ -1910,26 +1910,65 @@ export class NVIMPluginController implements vscode.Disposable {
         }
         const conf = this.highlightProvider.getDecoratorOptions(decorator);
         const options: vscode.DecorationOptions[] = [];
+        // Keep track of where VSCode will attach our decorations
+        const drawnAt = new Map();
         for (const [lineStr, cols] of decorations) {
             try {
                 const lineNum = parseInt(lineStr, 10) - 1;
                 const line = editor.document.lineAt(lineNum).text;
 
                 for (const [colNum, text] of cols) {
-                    // vim sends column in bytes, need to convert to characters
-                    // const col = colNum - 1;
-                    const col = Utils.convertByteNumToCharNum(line, colNum - 1);
-                    const opt: vscode.DecorationOptions = {
-                        range: new vscode.Range(lineNum, col, lineNum, col),
-                        renderOptions: {
-                            before: {
-                                ...conf,
-                                ...conf.before,
-                                contentText: text,
+                    if (Utils.getEasymotionOnTop()) {
+                        // 8 seems to be a magic number. It works for me, but I'm not sure why...
+                        const width = text.length;
+                        // vim sends column in bytes, need to convert to characters
+                        const col = Utils.convertByteNumToCharNumOnTop(line, colNum);
```

Comment:
No need for another function - `convertByteNumToCharNum()` is used now only for decorations. Also check how does it work with tabs